lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8332,6 +8348,13 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
                  swiftAttr->getAttribute());
       }
     }
+
+    if (seenUnsafe ||
+        importAsUnsafe(SwiftContext, dyn_cast<clang::RecordDecl>(ClangDecl),
```

Comment:
Minor: I think we should `cast<>` instead of `dyn_cast<>` here in order to get an assertion failure if `ClangDecl` isn't a C struct, since the callee won't handle nullptr correctly.